zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Schema.scala

```diff
@@ -39,42 +39,93 @@ object Relation extends Enumeration {
   val ~= = Value("~=")
 }
 
-class Field(val name: String, val dataType: DataType, val isOptional: Boolean = false) {
+trait Field {
+  val name: String
+  val dataType: DataType
+  val isOptional: Boolean = false
+
+  def as(name: String): Field
```

Comment:
I got the idea. Still I feel it shouldn't be a function of a class. 
As I proposed earlier, it will semantically make more sense to make it inside the `object Field` as a static function.